src/runtime/device_interface.cpp

```diff
@@ -237,6 +237,10 @@ WEAK int halide_device_free(void *user_context, struct buffer_t *buf) {
     debug(user_context) << "halide_device_free: " << buf
                         << " buf dev " << buf->dev
                         << " interface " << halide_get_device_interface(dev_field) << "\n";
+    if (buf->dev_dirty) {
```

Comment:
I thought about it more, this warning isn't useful. Any buffer that is just an intermediate between two device pipelines would trigger this. Sorry for the churn.